odl/discr/grid.py

```diff
@@ -1152,36 +995,23 @@ def __getitem__(self, slc):
                 new_maxpt.append(cvec[last])
                 new_shape.append(num)
 
-        return RegularGrid(new_minpt, new_maxpt, new_shape,
-                           as_midp=self.as_midp)
+        return RegularGrid(new_minpt, new_maxpt, new_shape)
 
     def __repr__(self):
-        """g.__repr__() <==> repr(g)."""
-        if self.as_midp:
-            return 'RegularGrid({}, {}, {}, as_midp=True)'.format(
-                list(self.min_pt), list(self.max_pt), list(self.shape))
-        else:
-            return 'RegularGrid({}, {}, {})'.format(
-                list(self.min_pt), list(self.max_pt), list(self.shape))
+        """Return ``repr(self)``."""
+        inner_str = '{}, {}, {}'.format(list(self.min_pt), list(self.max_pt),
+                                        tuple(self.shape))
```

Comment:
Numpy always returns a tuple as shape (even `np.shape(array_like)`), so I think it would make sense to hard-wire that connection, including `repr`.